plugin/src/main/groovy/com/btkelly/gnag/reporters/ViolationDetectorFactory.java

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2016 Bryan Kelly
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.btkelly.gnag.reporters;
+
+import com.btkelly.gnag.extensions.GnagPluginExtension;
+import com.btkelly.gnag.utils.ProjectHelper;
+
+import org.gradle.api.Project;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class ViolationDetectorFactory {
```

Comment:
These additional tasks have now somewhat mixed the definitions of `Task` vs `ViolationDetector`. I feel like if we go down the road of breaking out separate tasks we should get rid of the idea of a violation detector all together because they can now just be their own tasks.

Reply:
Sorry, I was busy lately and couldn't keep up with this. Can you please explain your concern here bit more deeply? Currently I am not seeing why the `ViolationDetector`s could be separate tasks, as then what the existing tasks would do?